### DISCUSSION TITLE
Slippage

### DISCUSSION THREAD

**Discussion by ghost** [edtechre/pybroker#169] [Q&A]
How do I make sure the random slippage works when I set price type? Only seems to work on the stoploss.

I can't enter a trade on the close of the signal bars so i need to set to the open of the next bar but I'm getting fills at the open price with no slippage which i don't think would reflect a true simulation.

I've checked against the price data using the trades and they are the same

if not long_pos and cross_up:

      target_shares = 100

      ctx.buy_shares = target_shares
      ctx.buy_fill_price = PriceType.OPEN
      ctx.hold_bars = 12
      ctx.stop_loss = ATR

elif long_pos and cross_down:
  ctx.sell_all_shares()
  ctx.sell_fill_price = pybroker.common.PriceType.OPEN

strategy = Strategy(data_source_M30, start_date, end_date,  config)

strategy.add_execution(exec_fn, tickers, indicators=list(indicators.values()))

strategy.set_slippage_model(RandomSlippageModel(min_pct=5.0, max_pct=10.0))
result = strategy.backtest(warmup=260, calc_bootstrap=True)

result.trades

**Comment by edtechre** [comment]
Hi @markxbaker,

The RandomSlippageModel only applies slippage to the number of shares filled, not price.

You can implement your own PriceSlippageModel though by extending [SlippageModel](https://www.pybroker.com/en/latest/reference/pybroker.slippage.html#pybroker.slippage.SlippageModel) and setting ``ctx.sell_price``.
